File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/06_Nested_Loops_Exercise/03_Sum_Prime_Non_Prime.js
function sumPrimeNonPrime(input) {
    let sumPrimes = 0;
    let sumNonPrimes = 0;

    let i = 0;
    while (true) {
        let currentCommand = input[i];
        i++;

        if (currentCommand == "stop") {
            break;
        }

        let currentNumber = Number(currentCommand);

        if (currentNumber < 0) {
            console.log('Number is negative.');
            continue;
        }


        let isPrime = true;
        for (let j = 2; j <= (currentNumber - 1); j++) {
            if (currentNumber % j == 0) {
                isPrime = false;
                break;
            }
        }

        if (isPrime == true) {
            sumPrimes += currentNumber;
        } else {
            sumNonPrimes += currentNumber;
        }
    }


    console.log(`Sum of all prime numbers is: ${sumPrimes}`);
    console.log(`Sum of all non prime numbers is: ${sumNonPrimes}`);

}


// sumPrimeNonPrime(["3",
//     "9",
//     "0",
//     "7",
//     "19",
//     "4",
//     "stop"]);

// sumPrimeNonPrime(["30",
//     "83",
//     "33",
//     "-1",
//     "20",
//     "stop"]);

File: Python_Courses/Python_Basics/Python_PB_2023_04/06_Nested_Loops_Exercise/03_Sum_Prime_Non_Prime.py
sum_primes = 0
sum_non_primes = 0

command = input()  # "stop" OR number as string
while not command == "stop":
    current_number = int(command)

    if current_number < 0:
        print("Number is negative.")
        command = input()
        continue

    is_prime = True
    for num in range(2, current_number):
        if current_number % num == 0:
            is_prime = False
            break

    if is_prime:
        sum_primes += current_number
    else:
        sum_non_primes += current_number

    command = input()

print(f"Sum of all prime numbers is: {sum_primes}")
print(f"Sum of all non prime numbers is: {sum_non_primes}")
